File: CS/IsFullExpandedOrCollapsed/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;

namespace IsFullExpandedOrCollapsed {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            new DevExpress.XtraGrid.Design.XViewsPrinting(gridControl1);
            gridView1.Columns["Discontinued"].Group();
            gridView1.Columns["Category"].Group();
            gridView1.OptionsView.NewItemRowPosition = NewItemRowPosition.Bottom;
            gridView1.OptionsView.ShowAutoFilterRow = true;
        }

        bool IsFullExpanded(GridView view) {
            if(view.GroupCount == 0) return true;
            if(view.DataRowCount == 0) return true;
            for(int i = -1; i > int.MinValue; i--) {
                if(!view.IsValidRowHandle(i)) return true;
                if(view.IsGroupRow(i) && !view.GetRowExpanded(i)) return false;
            }
            return true;
        }

        bool IsFullCollapsed(GridView view) {
            if(view.GroupCount == 0) return false;
            if(view.DataRowCount == 0) return false;
            for(int i = -1; i > int.MinValue; i--) {
                if(!view.IsValidRowHandle(i)) return true;
                if(view.IsGroupRow(i) && view.GetRowExpanded(i)) return false;
            }
            return true;
        }

        private void simpleButton1_Click(object sender, EventArgs e) {
            MessageBox.Show(string.Format("Fully exapanded: {0}\nFully collapsed: {1}", IsFullExpanded(gridView1), IsFullCollapsed(gridView1)));
        }
    }
}

File: VB/IsFullExpandedOrCollapsed/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid

Namespace IsFullExpandedOrCollapsed
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Dim TempXViewsPrinting As DevExpress.XtraGrid.Design.XViewsPrinting = New DevExpress.XtraGrid.Design.XViewsPrinting(gridControl1)
			gridView1.Columns("Discontinued").Group()
			gridView1.Columns("Category").Group()
			gridView1.OptionsView.NewItemRowPosition = NewItemRowPosition.Bottom
			gridView1.OptionsView.ShowAutoFilterRow = True
		End Sub

		Private Function IsFullExpanded(ByVal view As GridView) As Boolean
			If view.GroupCount = 0 Then
				Return True
			End If
			If view.DataRowCount = 0 Then
				Return True
			End If
			For i As Integer = -1 To Integer.MinValue + 1 Step -1
				If (Not view.IsValidRowHandle(i)) Then
					Return True
				End If
				If view.IsGroupRow(i) AndAlso (Not view.GetRowExpanded(i)) Then
					Return False
				End If
			Next i
			Return True
		End Function

		Private Function IsFullCollapsed(ByVal view As GridView) As Boolean
			If view.GroupCount = 0 Then
				Return False
			End If
			If view.DataRowCount = 0 Then
				Return False
			End If
			For i As Integer = -1 To Integer.MinValue + 1 Step -1
				If (Not view.IsValidRowHandle(i)) Then
					Return True
				End If
				If view.IsGroupRow(i) AndAlso view.GetRowExpanded(i) Then
					Return False
				End If
			Next i
			Return True
		End Function

		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
			MessageBox.Show(String.Format("Fully exapanded: {0}" & Constants.vbLf & "Fully collapsed: {1}", IsFullExpanded(gridView1), IsFullCollapsed(gridView1)))
		End Sub
	End Class
End Namespace
